File: sta308_final.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon May  9 17:18:30 2022

STA 308 - Final

Writing a program to do calculate the coefficient of variation in the 
change in unemployment rates by regions within the US

@author: joshuaroach
"""

import pandas as pd

## Loading in the base datasets
regions = pd.read_csv("https://tjfisher19.github.io/data/censusRegions.csv")
states = pd.read_csv("https://tjfisher19.github.io/data/state_abb_codes.csv")
unemployment = pd.read_csv("https://tjfisher19.github.io/data/marchStateUnemployment.csv")

## Merging the state and region datasets
statesWithRegions = states.merge(regions, left_on='Code', right_on='State')
## Merging the new dataset with the unemployment dataset
unemploymentStates = statesWithRegions.merge(unemployment, left_on = 'State_x', right_on = 'State')

## Creating a new variable 'unemploymentChange' that calculate the 
##   change in unemployment rates by state
unemploymentStates['unemploymentChange'] = unemploymentStates['March_21'] - unemploymentStates['March_22']

## Filters out DC to keep only states in the dataframe
unemploymentStates = unemploymentStates[unemploymentStates['Code'] != 'DC']

## Creating the final dataframe that caclulates mean, standard deviation
##  and coefficient of variation on unemploymentChange by region
unemploymentRegion = pd.DataFrame(
    {
     "MeanUnemploymentChange": unemploymentStates.groupby('Region')['unemploymentChange'].mean(),
     "StandardDeviation": unemploymentStates.groupby('Region')['unemploymentChange'].std(),
     "CoefficientOfVariance": unemploymentStates.groupby('Region')['unemploymentChange'].std()/unemploymentStates.groupby('Region')['unemploymentChange'].mean()
     }
    )
unemploymentRegion

File: sta308_final.R
################
## STA 308 - Final in R
## Josh Roach

library(tidyverse)

## Reading in the original datasets
regions <- read_csv("censusRegions.csv")
states <- read_csv("state_abb_codes.csv")
unemployment <- read_csv("marchStateUnemployment.csv")

## Merging states and region datasets to assign each state
##    to their specific region
statesWithRegions <- merge(states, regions, by.x = "Code", by.y = "State")
## Merging the new states and region dataset with the unemployment
##    dataset to assign regions to each state in this set
##    to align the unemployment numbers by region.
unemploymentStates <- merge(unemployment, statesWithRegions, by.x = "State", by.y = "State")

## Removing DC and creating a variable to calculate the change
##    in unemployment rates from 2021 to 2022.
unemploymentStates <- unemploymentStates %>%
  filter(State != "District of Columbia") %>%
  mutate(unemploymentChange = March_21 - March_22)

## Calculating the mean, standard deviation, and 
##    coefficient of variation by region on the 
##    unemployment rate change.
unemploymentRegions <- unemploymentStates %>%
  group_by(Region) %>%
  summarize(meanUnemployment = mean(unemploymentChange),
            sdUnemployment = sd(unemploymentChange),
            cvUnemployment = sdUnemployment/meanUnemployment)

